torch/optim/sparse_adam.py

```diff
@@ -64,59 +64,48 @@ def step(self, closure=None):
                 loss = closure()
 
         for group in self.param_groups:
+            params_with_grad = []
+            grads = []
+            exp_avgs = []
+            exp_avg_sqs = []
+            state_steps = []
```

Comment:
that's a lot of memory that is cumulated: one item per element of group['params']. is there any other way?

Reply:
as you pointed out, adam also does this since #44791

Reply:
as discussed with @iramazanli, `state = self.state[p]` means the memory is saved anyway.

Reply:
it was a nice discussion :) thanks for pointing this and discussing with me.